yaml_edit/lib/src/editor.dart

```diff
@@ -449,7 +437,7 @@ class YamlEditor {
           updatedYamlMap(parentNode, (nodes) => nodes.remove(keyOrIndex));
     }
 
-    _performEdit(edit, collectionPath, expectedNode);
+    _performEdit(edit!, collectionPath, expectedNode);
```

Comment:
Still not sure if this is the way to go but as you said, this never seems to be null.

Reply:
You can also use `late SourceEdit;`